policy/compression.go

```diff
@@ -0,0 +1,146 @@
+package policy
+
+import (
+	"errors"
+	"sync"
+
+	"github.com/m3db/m3cluster/kv"
+	"github.com/m3db/m3cluster/kv/util/runtime"
+	"github.com/m3db/m3metrics/generated/proto/schema"
+)
+
+var (
+	errNilValue              = errors.New("nil value received")
+	errNilPolicies           = errors.New("policies cannot be nil")
+	errNilCompressedPolicy   = errors.New("compressed policy cannot be nil")
+	errNilCompressedPolicies = errors.New("compressed policies cannot be nil")
+	errNilInternalPolicy     = errors.New("internal policy for compressed policy cannot be nil")
+)
+
+// CompressionMap is a mapping from Policy to integer.
+type CompressionMap map[Policy]int64
+
+// NewCompressionMap constructs a new CompressionMap from a slice of
+// compressed policies.
+func NewCompressionMap(policies []*schema.CompressedPolicy) (CompressionMap, error) {
```

Comment:
The ctor can probably just take in `ActivePolicies` as the parameter? We are probably gonna be watching it in kv anyway